src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Polly;
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Http
+{
+    public class PolicyHttpMessageHandler : DelegatingHandler
```

Comment:
The key building block here is a message handler that plugs in a policy to the pipeline. Ideally this is as generic as possible.

Not seen here: we'll want logging as well as that's one of the best ways that us building the integration adds value. I'd like to defer discussions about that until after the initial API for configuring clients is designed.

Reply:
:+1: to this highly general `PolicyHttpMessageHandler`

Reply:
@reisenberger - do you think that we'll end up having a generic and non-generic policy handler?  That seems logical to me based on the desire to provide both:
```
public static IHttpClientBuilder AddPolicyHandler(this IHttpClientBuilder builder, IAsyncPolicy policy)
public static IHttpClientBuilder AddPolicyHandler(this IHttpClientBuilder builder, IAsyncPolicy<HttpResponseMessage> policy)
```

Do you agree with this direction?

Reply:
Yes.

While only pure `IAsyncPolicy<HttpResponseMessage>` policies would be needed to govern `SendAsync(...)`, common Polly syntax (for example `Policy.TimeoutAsync(10)`) results more naturally in a non-generic `IAsyncPolicy`, so not supporting `IAsyncPolicy` might frustrate users.  

Generic policies are marginally more performant but any difference is in the order of microseconds or less.